src/app/beer_garden/garden.py

```diff
@@ -238,19 +247,199 @@ def create_garden(garden: Garden) -> Garden:
     }
     defaults = get_connection_defaults()
 
-    if garden.connection_params is None:
-        garden.connection_params = {}
-    garden.connection_params.setdefault("http", {})
+    return {config_map[key]: defaults[key] for key in config_map}
+
+
+def _validate_http_connection_params(conn_params):
```

Comment:
Here starts the code to cleanup bad connection configuration in Gardens (many of the changes in this file are related to renaming brewtils objects).